addons/website_mail/controllers/main.py

```diff
@@ -76,55 +75,50 @@ class WebsiteMail(http.Controller):
     @http.route(['/website_mail/follow'], type='json', auth="public", website=True)
     def website_message_subscribe(self, id=0, object=None, message_is_follower="on", email=False, **post):
         # TDE FIXME: check this method with new followers
-        cr, uid, context = request.cr, request.uid, request.context
-
-        partner_obj = request.registry['res.partner']
-        user_obj = request.registry['res.users']
+        Partner = request.env['res.partner'].sudo()
 
         _id = int(id)
         _message_is_follower = message_is_follower == 'on'
-        _object = request.registry[object]
+        _object = request.env[object]
 
         # search partner_id
         public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_ids = [user_obj.browse(cr, uid, uid, context).partner_id.id]
+        if request.env.uid != public_id:
+            partner_ids = [request.env.user.partner_id.id]
         else:
             # mail_thread method
-            partner_ids = _object._find_partner_from_emails(
-                cr, SUPERUSER_ID, [_id], [email], context=context, check_followers=True)
+            partner_ids = _object.sudo()._find_partner_from_emails([email], check_followers=True)
             if not partner_ids or not partner_ids[0]:
                 name = email.split('@')[0]
-                partner_ids = [partner_obj.create(cr, SUPERUSER_ID, {'name': name, 'email': email}, context=context)]
-
+                partner_ids = [Partner.create({'name': email, 'email': email}).id]
         # add or remove follower
+        object_record = _object.sudo().browse(_id)
         if _message_is_follower:
-            _object.check_access_rule(cr, uid, [_id], 'read', context)
-            _object.message_unsubscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+            object_record.check_access_rule('read')
+            object_record.message_unsubscribe(partner_ids)
             return False
         else:
-            _object.check_access_rule(cr, uid, [_id], 'read', context)
+            object_record.check_access_rule('read')
             # add partner to session
             request.session['partner_id'] = partner_ids[0]
-            _object.message_subscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+            object_record.message_subscribe(partner_ids)
             return True
```

Comment:
May be you can improve the code of website_message_subscribe(...) to something like,
--
_id = int(id)
        _message_is_follower = message_is_follower == 'on'
        _object_record = request.env[object].browse(_id)

        # search partner_id
        if request.env.uid != request.website.user_id.id:
            partner_ids = [request.env.user.partner_id.id]
        else:
            # mail_thread method
            partner_ids = _object_record.sudo()._find_partner_from_emails([email], check_followers=True)
            if not partner_ids or not partner_ids[0]:
                name = email.split('@')[0]
                partner_ids = [request.env['res.partner'].sudo().create({'name': name, 'email': email}).id]
        # add or remove follower
        if _message_is_follower:
            _object_record.check_access_rule('read')
            _object_record.sudo().message_unsubscribe(partner_ids)
            return False
        else:
            _object_record.check_access_rule('read')
            # add partner to session
            request.session['partner_id'] = partner_ids[0]
            _object_record.sudo().message_subscribe(partner_ids)
            return True
--